depends/toolchain.cmake.in

```diff
@@ -0,0 +1,135 @@
+set(CMAKE_SYSTEM_NAME @CMAKE_SYSTEM_NAME@)
+set(CMAKE_SYSTEM_PROCESSOR @CMAKE_SYSTEM_PROCESSOR@)
+
+function(split_compiler_launcher env_compiler launcher compiler)
+  set(${launcher})
+  list(GET ${env_compiler} 0 start_token)
+  if(start_token STREQUAL "env")
+    set(${compiler})
+    set(env_arg_parsing TRUE)
+    foreach(token IN LISTS ${env_compiler})
+      if(env_arg_parsing)
+        list(APPEND ${launcher} ${token})
+        set(env_arg_parsing FALSE)
+        continue()
+      elseif(token STREQUAL "-u")
+        list(APPEND ${launcher} ${token})
+        set(env_arg_parsing TRUE)
+        continue()
+      endif()
+      list(APPEND ${compiler} ${token})
+    endforeach()
+  else()
+    set(${compiler} ${${env_compiler}})
+  endif()
+  set(${launcher} ${${launcher}} PARENT_SCOPE)
+  set(${compiler} ${${compiler}} PARENT_SCOPE)
+endfunction()
+
+if(NOT CMAKE_C_COMPILER)
```

Comment:
Could you give a hint as to what's going on here? I'm a little lost :)

Reply:
The goal is to handle every "compiler invocation command", which we have in our build and test environment, properly. Including such cases as follows:
- `make -C depends CC=clang CXX='clang++ -stdlib=libc++'`, see [`ci/test/00_setup_env_native_nowallet_libbitcoinkernel.sh`](https://github.com/bitcoin/bitcoin/blob/master/ci/test/00_setup_env_native_nowallet_libbitcoinkernel.sh)
- `env -u LIBRARY_PATH clang`, see [`depends/hosts/darwin.mk`](https://github.com/bitcoin/bitcoin/blob/master/depends/hosts/darwin.mk), currently causes https://github.com/bitcoin/bitcoin/issues/21552

The main part is done in the `split_compiler_launcher()` function, which splits `env -u LIBRARY_PATH clang` in `env -u LIBRARY_PATH` and `clang`.

Rest of code makes it compatible with the supported range of CMake versions.

Reply:
> Won't this prohibit the use of `CMAKE_CXX_COMPILER_LAUNCHER` for ccache?

The future use of `CMAKE_{C,CXX}_COMPILER_LAUNCHER` with ccache can be peeked in https://github.com/bitcoin/bitcoin/pull/25797/commits/6f48c340a7cf9371dfa2931361c9cac0148cf005.